middlewares/accesslog/json_log_formatter.go

```diff
@@ -0,0 +1,235 @@
+package accesslog
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/types"
+)
+
+type tuple struct {
+	sourceKey, translatedKey string
+}
+
+type jsonLogFormatter struct {
+	timeFormat        string
+	coreMapping       []tuple
+	requestMapping    []tuple
+	originMapping     []tuple
+	downstreamMapping []tuple
+}
+
+// convertFieldsToMappings splits all the input strings on the colon character
+// and returns the {sourceKey, translatedKey} tuples. Any input string without a
+// colon will produce a tuple in which the sourceKey and translatedKey are the same.
+func convertFieldsToMappings(fields []string) []tuple {
+	var mapping []tuple
+	for _, s := range fields {
+		colon := strings.IndexByte(s, ':')
+
+		var t tuple
+
+		if colon > 0 {
+			t = tuple{
+				strings.TrimSpace(s[:colon]),
+				strings.TrimSpace(s[colon+1:]),
+			}
+		} else {
+			ws := strings.TrimSpace(s)
+			t = tuple{ws, ws}
+		}
+
+		mapping = append(mapping, t)
+	}
```

Comment:
This slight variation of my previous code snippet (this time pleasing the compiler) seems to work -- at least, the tests continue to go green:

```go
fields := strings.SplitN(s, ":", 2)
t := tuple{sourceKey: strings.TrimSpace(fields[0]), translatedKey: strings.TrimSpace(fields[0])}
if len(fields) == 2 {
  t.translatedKey = strings.TrimSpace(fields[1])
}
```

Would this work, or am I missing something?